app/[showStatus]/[showNameAndSeason]/scoring/page.tsx

```diff
@@ -0,0 +1,77 @@
+import { getCompetingEntityList } from "../../../utils/wikiQuery"
+import { getWikipediaContestantDataFetcher } from '../../../utils/wikiFetch'
+import generateListOfContestantRoundLists from '../../../generators/contestantRoundListGenerator'
+
+// This forces Next to only generate routes that exist in generateStaticParams, otherwise throw a 404
+export const dynamicParams = false
+
+interface showProperties {
+  showNameAndSeason: string,
+  showStatus: string
+}
+
+// Creates routes for scoring
+export function generateStaticParams() {
+  
+    // Necessary Node modules to fetch data
+    const fs = require('fs');
+    const path = require('path');
+    
+    // Based on availability in leagueData
+    const pathToLeagueData = path.join(process.cwd(), 'app', 'leagueData');
+    const shows:Array<showProperties> = [];
+    fs.readdirSync(pathToLeagueData).map((file: string) => {
+      // Needed status for url
+      const { LEAGUE_STATUS } = require(`../../../leagueConfiguration/${file.replace('.js', '.tsx')}`);
+      // Parses filename and converts it to url format
+      const showAndSeason = file.split('_');
+      const showNameFormatted = showAndSeason[0].split(/(?<![A-Z])(?=[A-Z])/).join('-').toLowerCase();
+      const showSeason = showAndSeason[1].replace('.js', '');
+      const showNameAndSeason = `${showNameFormatted}-${showSeason}`;
+      // Exporting properties as params
+      const showPropertiesObj = {
+        showNameAndSeason,
+        showStatus: LEAGUE_STATUS
+      }
+      shows.push(showPropertiesObj);
+    });
+    return shows;
+}
+
+export default async function Scoring({ params }: {
+    params: Promise<{ showStatus: string; showNameAndSeason: string }>
+  }) {
+    // Wait for parsing and retrieving params
+    const { showNameAndSeason } = await params;
+    // Formatting to file naming convention
+    const showAndSeasonArr = showNameAndSeason.split('-');
+    const showSeason = showAndSeasonArr[showAndSeasonArr.length - 1];
+    showAndSeasonArr.pop();
+    const showName = showAndSeasonArr.map((word) => word.charAt(0).toUpperCase() + word.slice(1)).join('');
+    const fileName = `${showName}_${showSeason}`;
+    // "Dynamically" (still static site generated) retrieving modules
+    const { CONTESTANT_LEAGUE_DATA } = await require(`../../../leagueData/${fileName}.js`);
+    const { WIKI_API_URL, GOOGLE_SHEET_URL, CAST_PHRASE } = await require(`../../../leagueConfiguration/${fileName}.tsx`);
+
+    const dataFetcher = getWikipediaContestantDataFetcher(WIKI_API_URL, CAST_PHRASE);
+    let listOfContestantRoundLists;
+
+    // Check for Amazing Race due to additional param
+    if(showName.match('AmazingRace')){
+      listOfContestantRoundLists = await generateListOfContestantRoundLists(dataFetcher, CONTESTANT_LEAGUE_DATA)
+    } else {
+      listOfContestantRoundLists = await generateListOfContestantRoundLists(dataFetcher, CONTESTANT_LEAGUE_DATA, getCompetingEntityList)
+    }
```

Comment:
fft: I don't like this 🤮 I would like to find a way to not do this!

Reply:
Neither would I. We will fix this in a follow up PR